### ISSUE TITLE
A11y keyboard focus control in the Icon Component

### ISSUE THREAD

**Issue by DaveBest99** [angular/material2#3707]
#### Bug, feature request, or proposal:
Bug: Screen reader cannot find the icon.

#### What is the expected behavior?
Icons and images are expected to have a descriptive text associated with them.

#### What is the current behavior?
The SVG icon image is invisible to the screen reader.

#### What are the steps to reproduce?
See below.

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

#### What is the use-case or motivation for changing an existing behavior?
Create a good screen reader user experience.

#### Which versions of Angular, Material, OS, browsers are affected?
Using screen reader JAWS17 with browsers Firefox, Internet Explorer 11, and Google Chrome, on Windows10.

#### Is there anything else we should know?
Screen reader user experience follows:

1. Press Tab key to the Icon demo component in the Tab Panel.
The screen reader focus skips over the Tab Panel and moves directly to the page footer. The SVG icons component is completely invisible to blind users, as the Thumbs-Up image has no Alt-Text, aria-label, or text description.
Note, according to the "Overview" explanation:
If an aria-label attribute is set on the md-icon element, its value will be used as-is. If not, the md-icon component will attempt to set the aria-label
value from one of these sources:
- The alt attribute
- The fontIcon input
- The name of the icon from the svgIcon input (not including any namespace)
- The text content of the component (for ligature icons)

See:
Accessibility - Usability - Material design guidelines
https://material.io/guidelines/usability/accessibility.html#

**Comment by jelbourn** [comment]
Closing this since we now have appropriate guidance for using icons in general.
